components/database/postgresql-14/Makefile

```diff
@@ -0,0 +1,130 @@
+CONFIGURE_OPTIONS += --docdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/doc
+CONFIGURE_OPTIONS += --htmldir=/usr/postgres/$(COMPONENT_MJR_VERSION)/doc
+CONFIGURE_OPTIONS += --enable-nls
+CONFIGURE_OPTIONS += --with-system-tzdata=/usr/share/lib/zoneinfo
+CONFIGURE_OPTIONS.64 += --with-tcl
+CONFIGURE_OPTIONS.64 += --with-python
+CONFIGURE_OPTIONS.64 += --with-pam
+CONFIGURE_OPTIONS += --with-openssl
+CONFIGURE_OPTIONS += --with-libedit-preferred
+CONFIGURE_OPTIONS += --with-libxml
+CONFIGURE_OPTIONS += --with-libxslt
+CONFIGURE_OPTIONS += --with-lz4
+CONFIGURE_OPTIONS += --with-gssapi
+CONFIGURE_OPTIONS += --enable-thread-safety
+CONFIGURE_OPTIONS += --enable-dtrace
+CONFIGURE_OPTIONS += --enable-integer-datetimes
+CONFIGURE_OPTIONS += --with-includes=/usr/include
+CONFIGURE_OPTIONS += --with-tclconfig=/usr/lib
+CONFIGURE_OPTIONS.64 += --with-perl
+CONFIGURE_OPTIONS += --with-uuid=e2fs
+
+CONFIGURE_OPTIONS.32 += --bindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin/$(MACH32)
+CONFIGURE_OPTIONS.32 += --libexecdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin/$(MACH32)
+CONFIGURE_OPTIONS.32 += --sbindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin/$(MACH32)
+CONFIGURE_OPTIONS.32 += --libdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/lib
+
+CONFIGURE_OPTIONS.64 += --bindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin
+CONFIGURE_OPTIONS.64 += --libexecdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin
+CONFIGURE_OPTIONS.64 += --sbindir=/usr/postgres/$(COMPONENT_MJR_VERSION)/bin
+CONFIGURE_OPTIONS.64 += --libdir=/usr/postgres/$(COMPONENT_MJR_VERSION)/lib/$(MACH64)
+
+CONFIGURE_OPTIONS+=$(CONFIGURE_OPTIONS.$(BITS))
+
+CONFIGURE_ENV	+= PYTHON=$(PYTHON)
+CONFIGURE_ENV.32     += PERL=$(PERL)
+CONFIGURE_ENV.64     += PERL=$(PERL.5.24)
+
+COMPONENT_BUILD_ENV += MAKELEVEL=0
+COMPONENT_BUILD_ENV += PATH=$(PATH)
+COMPONENT_INSTALL_ENV += PATH=$(PATH)
+
+COMPONENT_BUILD_TARGETS = world
+COMPONENT_INSTALL_TARGETS = install-world
+
+COMPONENT_POST_INSTALL_ACTION.64= cd $(BUILD_DIR_$(BITS))/src/test/regress; env $(COMPONENT_INSTALL_ENV) gmake $(COMPONENT_INSTALL_ARGS) install-lib install-tests;
+
+COMPONENT_POST_INSTALL_ACTION+= $(COMPONENT_POST_INSTALL_ACTION.$(BITS))
+COMPONENT_POST_INSTALL_ACTION+= cd $(PROTO_DIR) &&  \
+             $(MV) usr/postgres/$(COMPONENT_MJR_VERSION)/include/pg_config.h usr/postgres/$(COMPONENT_MJR_VERSION)/include/pg_config-$(BITS).h  && \
+             $(MV) usr/postgres/$(COMPONENT_MJR_VERSION)/include/server/pg_config.h usr/postgres/$(COMPONENT_MJR_VERSION)/include/server/pg_config-$(BITS).h
+
+ENV=/usr/bin/env -i
+
+COMPONENT_TEST_ENV += PATH=$(PATH)
+
+COMPONENT_TEST_TRANSFORMS= \
+	'-n ' \
+	'-e "s:[ \t]*[0-9]* ms::"' \
+	'-e "/ok/p"' \
+	'-e "/FAIL/p"' \
+	'-e "/failed/p"'
+
+
+REQUIRED_PACKAGES += SUNWcs
+REQUIRED_PACKAGES += library/libedit
+REQUIRED_PACKAGES += library/libxml2
+REQUIRED_PACKAGES += library/libxslt
+REQUIRED_PACKAGES += library/lz4
+REQUIRED_PACKAGES += library/security/openssl-11
+REQUIRED_PACKAGES += library/zlib
```

Comment:
Did you change the openssl mediator on your system? When I try to publish here I'll get lots of dependency errors like this:
oi-userland/components/database/postgresql-14/build/manifest-i386-postgresql14-contrib.depend has unresolved dependency '
    depend type=require fmri=__TBD pkg.debug.depend.file=libcrypto.so.1.0.0 \
        pkg.debug.depend.reason=usr/postgres/14/bin/pg_verifybackup \
        pkg.debug.depend.type=elf \
        pkg.debug.depend.path=lib/64 \
        pkg.debug.depend.path=usr/gcc/7/lib/amd64 \
        pkg.debug.depend.path=usr/lib/64 \
        pkg.debug.depend.path=usr/postgres/14/lib/amd64'.

You'll probably need to tweak the CONFIGURE_OPTIONS to let it find openssl-1-1 as USE_OPENSSL11=yes doesn't seem to be enough.

Reply:
I didn't think about it but yes it seems I have done so ages ago.
e.g.

openssl             local     1.1     system     openssl

I do wonder why by today things are still not on 1.1 by default given that 3 is the new stable by now.

I already spend a whole day on this so not sure when and if I have more time to spend given that building packages
on OpenIndiana really is painfull compared to Omnios. I think the whole hand building of p5m files is getting somewhat
tiresome given the big changes in packages these days. Maybe postgresql is even worse given it has such a split set
of p5m files.

Reply:
Short answer: Not enough time/volunterrs
Long answer: We still have some packages that need to be changed to use openssl-1.1. Some of these packages need non-obvious tweaks, some need to be updated to support openssl-1.1 and some cannot be changed or updated because there is no update available.

Reply:
On the long answer you could opt to change to 1.1 default and fix the stuff that doesn't work which is probably a better approach (e.g. less work ) then putting the burden on all new stuff. And maybe you need to freeze and obsolete some stuff that doesn't work with openssl 1.1. I don't use postgresql on OpenIndiana only wanted to give people the same 
and more performance DISM solution. But I kind of get why no one bothered with pg13 or pg14.

Reply:
The issue is that consumers of the broken packages will likely not be happy... Often it is just a matter of importing a few patches. I have fixed myself some of these packages although I literally do not care about them.

Reply:
Changing the mediator doesn't break the old packages as they will never be rebuild anyway. It seems the configure from postgresql doesn't give you the option to specify a with value setting the INC or LIB dir. other option would be to compile against openssl-1.0. I'll give that a try now and update the deps then someone can fix things who has to much time to spend or sees a simple way to do it.

Reply:
Ok with openssl 1.0 this time build and published with mediator set to openssl 1.0